### ISSUE TITLE
Why do we add 2 in LSTM logsumexp?

### ISSUE THREAD

**Issue by tomjaguarpaw** [awf/ADBench#143]
LSTM logsumexp has [2 added to the sum](https://github.com/awf/ADBench/blob/5020acbcd3948aab74be042a9cbaf670bb28b21c/src/cpp/shared/lstm.h#L22).  Do we know why this is?  The earliest occurrence I can see is https://github.com/awf/ADBench/commit/d75f713d8a6fa4f8ff3f988692c1a5c76238abe9#diff-cadabfe323e7f3ac5fe22b5de628d2aeR24

**Comment by awf** [comment]
Yes that is very odd, and doesn't ring a bell.   Let's also make this issue track other LSTM oddnesses: diagonal weight matrix.  Any more?
